pkg/metrics/server.go

```diff
@@ -0,0 +1,66 @@
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"k8s.io/klog"
+)
+
+var (
+	tlsCRT = "/etc/secrets/tls.crt"
+	tlsKey = "/etc/secrets/tls.key"
+)
+
+// RunServer starts the metrics server.
+func RunServer(port int, stopCh <-chan struct{}) {
+	if port <= 0 {
+		klog.Error("invalid port for metric server")
+		return
+	}
+
+	bindAddr := fmt.Sprintf(":%d", port)
+	router := http.NewServeMux()
+	router.Handle("/metrics", promhttp.Handler())
+	srv := &http.Server{
+		Addr:    bindAddr,
+		Handler: router,
+	}
+
+	go func() {
+		err := srv.ListenAndServeTLS(tlsCRT, tlsKey)
+		if err != nil && err != http.ErrServerClosed {
+			klog.Errorf("error starting metrics server: %v", err)
+		}
+	}()
+	<-stopCh
+	if err := srv.Close(); err != nil {
+		klog.Errorf("error closing metrics server: %v", err)
+	}
+}
+
+// ConfiguredStorage sets the metric that indicates if we have or not
+// a configured storage for the image registry.
+func ConfiguredStorage(configured bool) {
+	if configured {
+		storageConfigured.Set(1)
+		return
+	}
+	storageConfigured.Set(0)
+}
+
+// Degraded sets the metric that indicates if the operator is in degraded
+// mode or not.
+func Degraded(deg bool) {
+	if deg {
+		degraded.Set(1)
+		return
+	}
+	degraded.Set(0)
+}
+
+// IncStorageChanged increases the number of times the storage got changed.
```

Comment:
The description doesn't match the current implementation.

Currently this function is called when the storage driver creates a new storage medium. Is it intentionally?

Reply:
The idea here is to take care of how many times we "create" a storage medium and report if we ended up doing it multiple times. I have renamed the functions to reflect this but I would like to hear what are your concerns.

Reply:
@dmage I've already implemented the requested changes and due to inactivity I am resolving this comment.